js/course.js

```diff
@@ -92,6 +92,26 @@ function setCourseOptionsContent(modal, options) {
             }
         }
     }
+
+    // Add request icon button only if default icon doesn't already exist
+    let hasDefaultIcon = (function (course) {
+        for (let iconPattern of icons) {
+            if (iconPattern.regex != "." && course.match(new RegExp(iconPattern.regex, 'i'))) {
+                return true;
+            }
+        }
+        return false;
+    })(options.courseName);
+    if (!hasDefaultIcon) {
+        modal.element.querySelector(".splus-modal-contents").appendChild(
+            createElement("div", ["setting-entry"], { id: "request-course-icon-wrapper" }, [
+                createElement("h2", ["setting-title"], { textContent: "Request Icon: " }, [
+                    createElement("input", [], { type: "button", id: "request-course-icon-button", value: "Request Icon", onclick: () => window.open(`https://docs.google.com/forms/d/e/1FAIpQLSe-v0vRE4Obwkx6iL37aztz3kmpqYBBVxKEsdVxu8CZqk1OCQ/viewform?entry.50164059=${options.courseName}`, "_blank") })

```

Comment:
Also, is it possible we could use an `a` here? (Probably just wrap the `input` in an `a`.) That way the user could e.g. middle click it and overall more conveniently interface with it as a link.

Reply:
I agree with all of your other comments, but changing this to an `a` would make its behavior differ from every other button in the Schoology interface, and in addition the link already has `target="_blank"` so the middle click scenario isn't super meaningful.

Reply:
Most other buttons aren't a static link - they're either dynamic or internal or need a JavaScript onClick. I propose this here because this one doesn't need any JavaScript, it could be a static `href`. It's nicer for the user, I contend, for predictable link-like behavior to be indicated by the signs of an `a`: hovering shows the URL, opening in new tab or window is easy (in my test environment, it opened in a new window when I clicked on it but a new tab, which I prefer, when I middle-clicked it - not sure why, but you see the point), etc. So I would say there is a small but meaningful difference to having an `a` here.

That said I see your point that `a` would be different and inconsistent. (I also know we have lots of fun dealing with Schoology's internal inconsistencies. That said, if we wrote a good CSS selector here, wrapping this might not matter, e.g. `.setting-entry .setting-title input`.) I think the behavioral difference from an `a` is noticeable and this button has properties (fixed, external HTTP link) that make it different, but that does necessitate a code difference (even though I don't see a concrete issue arising out of this, I respect the possibility). I'm not sure.

I'm still leaning towards `a`, but I'll leave it up to you: there are merits to either.

Reply:
Ok, I do see your point now. I think I'll just replace the button with a standard link to avoid confusion.